src/common.cc

```diff
@@ -111,4 +121,337 @@ Join(const std::vector<std::string>& vec, const std::string& delim)
   return ss.str();
 }
 
+#ifdef TRITON_BIG_ENDIAN
+
+/// Returns uint16_t input with byte order swapped
+///
+/// \param[in] input 
+/// \return input with byte order swapped
+///
+uint16_t
+bswap(const uint16_t input)
+{
+  return __builtin_bswap16(input);
+}
+
+/// Returns uint32_t input with byte order swapped
+///
+/// \param[in] input
+/// \return input with byte order swapped
+///
+uint32_t
+bswap(const uint32_t input)
+{
+  return __builtin_bswap32(input);
+}
+
+/// Returns uint64_t input with byte order swapped
+///
+/// \param[in] input
+/// \return input with byte order swapped
+///
+uint64_t
+bswap(const uint64_t input)
+{
+  return __builtin_bswap64(input);
+}
+
+/// Returns uint32_t length with byte order swapped and
+/// returns length in host byte order.
+///
+/// \param[in] length
+/// \param[in] host_byte_order whether original length is in host byte order
+/// \param[out] host_length length in host byte order
+/// \return length with byte order swapped
+///
+uint32_t
+bswap_length(
```

Comment:
Suggestion change all functions to:

```
SwapBytes()
SwapLength()
```
Instead of 
```
SwapEndian()
```

Partial would be via parameter overloading - or alternatively:

```
SwapBytes()
SwapLength()
SwapBytesPartial()
SwapLengthPartial()
```

Reply:
Looks good!